test/modern_bpf/test_suites/syscall_enter_suite/inotify_init_e.cpp

```diff
@@ -0,0 +1,42 @@
+#include "../../event_class/event_class.h"
+
+#if defined(__NR_inotify_init) && defined(__NR_close)
+TEST(SyscallEnter, inotify_initE)
+{
+
+	auto evt_test = new event_test(__NR_inotify_init, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	uint32_t flags = 15;
+	int32_t fd = syscall(__NR_inotify_init, flags);
+	assert_syscall_state(SYSCALL_SUCCESS, "inotify_init", fd, NOT_EQUAL, -1);
+	syscall(__NR_close, fd);
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: flags (type: PT_FLAGS8) */
+	evt_test->assert_numeric_param(1, (uint8_t)flags);
```

Comment:
hmm... so if inotify_init(void) does not process the flags, then the BPF program just passes the first argument simply thru? How does it look like for a simply `inotify_init()` call where the BPF program then reports the (whatever) content the register for the first argument contains? If so, could that be a security issue/information leak?

Reply:
Not sure if this is an `information leak` but it could be... In my tests, it always returns `0`, but obviously we are not sure about the content, we must fix it!

Reply:
> Not sure if this is an `information leak` but it could be...

It could be.... and depending on the architecture ELF ABI, argument registers might become call-clobbered/cleared etc.  However, being on the safe side is better.

Reply:
Thanks for [flags removal in `eventfd`](https://github.com/falcosecurity/libs/pull/516/commits/ad4a3d9a42b09b1f6e0ce5425b12ba5ca16cae2a) which also had the potential for a leak.